workspaces/src/network/account.rs

```diff
@@ -40,7 +101,188 @@ impl Contract {
         &self.account.id
     }
 
+    pub fn as_account(&self) -> &Account {
+        &self.account
+    }
+
     pub(crate) fn signer(&self) -> &InMemorySigner {
         self.account.signer()
     }
+
+    /// Call the current contract's function using the contract's own account
+    /// details to do the signing. Returns a Builder object that we will make
+    /// use to populate the rest of the call details.
+    ///
+    /// If we want to make use of the contract's account to call into a
+    /// different contract besides the current one, use
+    /// `contract.as_account().call` instead.
+    pub fn call<'a, T: Network>(
+        &self,
+        worker: &'a Worker<T>,
+        function: String,
+    ) -> CallBuilder<'a, T> {
+        self.account.call(worker, self.id().clone(), function)
+    }
+
+    /// Call a view function into the current contract. Returns a result that
+    /// yields a JSON string object.
+    pub async fn view<T: Network>(
+        &self,
+        worker: &Worker<T>,
+        function: String,
+        args: Vec<u8>,
+    ) -> anyhow::Result<String> {
+        worker.view(self.id().clone(), function, args).await
+    }
+
+    /// Deletes the current contract, and returns the execution details of this
+    /// transaction. The beneciary will receive the funds of the account deleted
+    pub async fn delete_contract<T: Network>(
+        self,
+        worker: &Worker<T>,
+        beneficiary_id: AccountId,
+    ) -> anyhow::Result<CallExecutionDetails> {
+        self.account.delete_account(worker, beneficiary_id).await
+    }
+}
+
+pub struct CallBuilder<'a, T> {
+    worker: &'a Worker<T>,
+    signer: InMemorySigner,
+    contract_id: AccountId,
+
+    function: String,
+    args: Option<Vec<u8>>,
+    deposit: Option<u128>,
+    gas: Option<u64>,
+}
+
+impl<'a, T: Network> CallBuilder<'a, T> {
+    fn new(
+        worker: &'a Worker<T>,
+        contract_id: AccountId,
+        signer: InMemorySigner,
+        function: String,
+    ) -> Self {
+        Self {
+            worker,
+            signer,
+            contract_id,
+            function,
+            args: None,
+            deposit: None,
+            gas: None,
+        }
+    }
+
+    pub fn with_args(mut self, args: Vec<u8>) -> Self {
+        self.args = Some(args);
+        self
+    }
+
+    pub fn with_deposit(mut self, deposit: u128) -> Self {
+        self.deposit = Some(deposit);
+        self
+    }
+
+    pub fn with_gas(mut self, gas: u64) -> Self {
+        self.gas = Some(gas);
+        self
+    }
+
+    pub async fn transact(self) -> anyhow::Result<CallExecutionDetails> {
+        self.worker
+            .client()
+            .call(
+                &self.signer,
+                self.contract_id,
+                self.function,
+                self.args
+                    .expect("required `with_args` to be specified as apart of Call"),
```

Comment:
A call function doesn't need to use params, does it? Why not just default to empty vec if it isn't required (I'm not 100% sure it isn't required, but I can't see why it would be)

Reply:
So actually, empty vec fails for (certain or maybe all) calls. We'd at least need empty brackets such as `serde_json::json!({})`, but not sure what the equivalent borsh would be though. I think we can just default to `{}` for now and force people to do `with_args_borsh` if they expect their args to be borsh serialized

Reply:
Wait, why is this a requirement? Is it with the API? There should not be this requirement. Args should be optional

Reply:
Actually there's an interesting case with this. Empty `Vec::new()` fails for functions taking all `Option`s, so we're required to supply `{}` here instead. Not sure why that is the case

Reply:
Also, just tested it purely with a function that takes no arguments, and it took both `Vec::new()` and `json!({})` without a problem. I think it's just how json rpc is working under the hood. We can take a closer look another time, but let's just go with a default of `json!({})` so it fits both cases. WDYT?

Reply:
it's fine, I just opened one up here: https://github.com/near/workspaces-rs/issues/36